lib/krikri/enrichments/iso_enrich.rb

```diff
@@ -0,0 +1,150 @@
+module Krikri::Enrichments
+  ##
+  # Converts text fields and/or providedLabels to ISO 689-3 URIs (Lexvo)
+  #
+  # Transforms text values matching either (English) labels or ISO language
+  # codes from Lexvo into DPLA::MAP::Controlled::Language resources with
+  # skos:exactMatch of the appropriate Lexvo URIs. Original string values are
+  # retained as dpla:providedLabel.
+  #
+  # @example matching string values
+  #   iso = IsoEnrich.new
+  #   iso.enrich_value('fin')     # matches 'http://lexvo.org/id/iso639-3/fin'
+  #   iso.enrich_value('finnish') # matches 'http://lexvo.org/id/iso639-3/fin'
+  #   iso.enrich_value('eng')     # matches 'http://lexvo.org/id/iso639-3/eng'
+  #   iso.enrich_value('english') # matches 'http://lexvo.org/id/iso639-3/eng'
+  #   iso.enrich_value('English') # matches 'http://lexvo.org/id/iso639-3/eng'
+  #   iso.enrich_value('English') # matches 'http://lexvo.org/id/iso639-3/eng'
+  #
+  # If no matches are found, returns a bnode with the input value as
+  # providedLabel.
+  #
+  # If passed an ActiveTriples::Resource, the enrichment will:
+  #
+  #   - Perform the above text matching on any present `providedLabel`s,
+  #     returning the original node if no results are found.  If multiple
+  #     values are provided and multiple matches found, they will be
+  #     deduplicated.
+  #   - Leave DPLA::MAP::Controlled::Language objects that are not bnodes
+  #     unaltered.
+  #   - Remove any values which are not either bnodes or members of
+  #     DPLA::MAP::Controlled::Language.
+  #
+  # @example matching node values
+  #   iso = IsoEnrich.new
+  #   lang = DPLA::MAP::Controlled::Language.new
+  #   lang.providedLabel = 'eng'
+  #   iso.enrich_value(lang)     # matches ['http://lexvo.org/id/iso639-3/fin']
+  #   lang.providedLabel = 'fin', 'eng'
+  #   iso.enrich_value(lang)     # matches ['http://lexvo.org/id/iso639-3/fin',
+  #                              #          'http://lexvo.org/id/iso639-3/eng']
+  #
+  # Label matches are cached within the enrichment instance.
+  #
+  # @see DPLA::MAP::Controlled::Language
+  # @see http://www.lexvo.org/
+  class IsoEnrich
```

Comment:
I'd rename the class something slightly clearer, e.g. `LanguageToISO639` - as is, the class name isn't obvious since ISO provides a number of standards we'll use in the application.